packages/gatsby/src/schema/build-node-types.js

```diff
@@ -192,33 +190,20 @@ module.exports = async ({ parentSpan }) => {
         name: typeName,
         type: gqlType,
         args: filterFields,
-        async resolve(a, args, context) {
+        async resolve(a, queryArgs, context) {
           const path = context.path ? context.path : ``
-          const runSift = require(`./run-sift`)
-          let latestNodes
-          if (
-            process.env.NODE_ENV === `production` &&
-            nodesCache.has(typeName)
-          ) {
-            latestNodes = nodesCache.get(typeName)
-          } else {
-            latestNodes = getNodesByType(typeName)
-            nodesCache.set(typeName, latestNodes)
-          }
```

Comment:
can we apply this caching to `getNodesByType` utility function so it doesn't filter all nodes every time in production builds?